Always use /proc/sys/kernel/random/boot_id to confirm reboot on Linux

ssiegel opened the pull request:
##### SUMMARY
/proc/sys/kernel/random/boot_id is available since kernel 2.3.16 and
should be safe to rely on.

The previously used method by checking the system boot time using who -b
turned out to be unreliable: Some systems lacking an RTC report the Unix
epoch as boot time, but the code trying to detect that did't always
work.

Fixes #46562
Fixes #47371

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
reboot.py

##### ANSIBLE VERSION
```paste below
ansible 2.7.0
  config file = None
  configured module search path = [u'/home/stefan/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/stefan/.local/venvs/ansible/lib/python2.7/site-packages/ansible
  executable location = /home/stefan/.local/bin/ansible
  python version = 2.7.14 (default, Oct 12 2017, 15:50:02) [GCC]
```

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

ssiegel committed:
Always use /proc/sys/kernel/random/boot_id to confirm reboot on Linux

/proc/sys/kernel/random/boot_id is available since kernel 2.3.16 and
should be safe to rely on.

The previously used method by checking the system boot time using who -b
turned out to be unreliable: Some systems lacking an RTC report the Unix
epoch as boot time, but the code trying to detect that did't always
work.

Closes #46562

samdoran commented:
Please create a changelog fragment. See [fragments](https://github.com/ansible/ansible/tree/stable-2.7/changelogs/fragments) for examples.

samdoran committed:
Change DEFAULT_BOOT_TIME_COMMAND

- change to usinsg /proc by default
- add BOOT_TIME_COMMANDS for BSD, Solaris, and macOS
